numpy/__init__.pyi

```diff
@@ -1,8 +1,91 @@
 # very simple, just enough to start running tests
+#
+import builtins
+from typing import Any, Mapping, List, Optional, Tuple, Union
 
-class ndarray: pass
+from numpy.core._internal import _ctypes
+
+class dtype:
+    alignment: int
+    byteorder: str
+    char: str
+    descr: List[Tuple[str, str]]
+    fields: Optional[Mapping[str, Union[Tuple[dtype, int], Tuple[dtype, int, str]]]]
+    flags: int
+    hasobject: bool
+    isbuiltin: int
+    isnative: bool
+    isalignedstruct: bool
+    itemsize: int
+    kind: str
+    name: str
+    names: Optional[Tuple[str, ...]]
+    num: int
+    shape: Tuple[int, ...]
+    ndim: int
+    subdtype: Optional[Tuple[dtype, Tuple[int, ...]]]
+
+
+    def newbyteoder(self, new_order:str = ...): dtype
+
+    str: builtins.str
+    type: builtins.type
+
+    @property
+    def base(self) -> dtype: ...
+
+
+_dtype = dtype      # for ndarray type
+
+
+class flagsobj:
```

Comment:
`flagsobj` is not exposed by NumPy's public API, so let's give it a private name (`_flagsobj`?)